### PR TITLE
17057 - Checking for Authentication Endpoint for Entities

### PR DESCRIPTION
**Pull request by rodrigo-barraza**
*Issue #:17057*
https://github.com/bcgov/entity/issues/17057

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Review by seeker25 — APPROVED** [review]

**Comment by seeker25** [comment]
Jumped the gun on that approve

File: auth-api/src/auth_api/resources/v1/entity.py [hunk 1]
```diff
@@ -128,6 +128,25 @@ def delete_entity(business_identifier):
...
+@_jwt.requires_auth
+def get_entity_authentication(business_identifier):
+    """Get passcode or password for the Entity identified by the provided business identifier."""
+    # This route allows public users to see if businesses have a form of authentication.
+    # It's used by the business dashboard for magic link.
+    if ((entity := EntityService.find_by_business_identifier(business_identifier, skip_auth=True)) and
+    (contact := entity.get_contact())):
+        has_valid_pass_code = (entity.pass_code_claimed == 'f' and entity.pass_code is not None) or entity.corp_type_code in ['SP','GP']
+        return {
+            'contactEmail': contact.email,
```
Comment (ochiu) [inline auth-api/src/auth_api/resources/v1/entity.py:143]: We should probably mask the email that comes back like we do everywhere else. There is an existing utility function for this.

`def mask_email(email: str) -> str:` (src/auth_api/utils/util.py)

@seeker25 what do you think?

File: auth-api/tests/unit/api/test_entity.py [hunk 2]
```diff
@@ -486,3 +486,16 @@ def test_get_entity_contacts(client, jwt, session):
     assert data['email'] == 'fo*@ba*****'
     assert 'phone' not in data
     assert 'phone_extension' not in data
+
+
+def test_get_entity_authentication(client, jwt, session):
+    """Assert that an entity authentication can be retrieved."""
+    headers = factory_auth_header(jwt=jwt, claims=TestJwtClaims.public_user_role)
+    rv = client.get(f'/api/v1/entities/{TestEntityInfo.entity1["businessIdentifier"]}/authentication',
```
Comment (ochiu) [inline auth-api/tests/unit/api/test_entity.py:494]: You will probably need to create the entity first, should be able to reference other tests for this

**Review by seeker25 — CHANGES_REQUESTED** [review]
Fix tests / lints

File: auth-api/src/auth_api/resources/v1/entity.py (hunk 1 above)
Comment (seeker25) [reply auth-api/src/auth_api/resources/v1/entity.py:143]: Yup, I may have missed this part

Comment (seeker25) [reply auth-api/src/auth_api/resources/v1/entity.py:143]: Need new service property in entity:
```
    @property
    def pass_code_claimed(self):
        """Return the pass_code_claimed for this entity."""
        return self._model.pass_code_claimed
```

rename `entity.corp_type_code` -> `entity.corp_type`

File: auth-api/tests/unit/api/test_entity.py (hunk 2 above)
Comment (seeker25) [reply auth-api/tests/unit/api/test_entity.py:494]: existing test has it:

```
def test_get_entity_contacts(client, jwt, session):
    """Assert that an entity contacts can be retrieved."""
    headers = factory_auth_header(jwt=jwt, claims=TestJwtClaims.system_role)
    rv_create = client.post('/api/v1/entities', data=json.dumps(TestEntityInfo.entity1),
                            headers=headers, content_type='application/json')
    assert rv_create.status_code == http_status.HTTP_201_CREATED
    client.post('/api/v1/entities/{}/contacts'.format(TestEntityInfo.entity1['businessIdentifier']),
                headers=headers, data=json.dumps(TestContactInfo.contact1), content_type='application/json')

    headers = factory_auth_header(jwt=jwt, claims=TestJwtClaims.public_user_role)
    rv = client.get(f'/api/v1/entities/{TestEntityInfo.entity1["businessIdentifier"]}/contacts',
                    headers=headers, content_type='application/json')
    assert rv.status_code == http_status.HTTP_200_OK
    data = json.loads(rv.data)
    assert data['email'] != TestContactInfo.contact1['email']
    assert data['email'] == 'fo*@ba*****'
    assert 'phone' not in data
    assert 'phone_extension' not in data
    ```

**Review by seeker25 — APPROVED** [review]

**Pull request merged**